domain/application/status.go

```diff
@@ -4,7 +4,27 @@
 package application
 
 import (
-	corestatus "github.com/juju/juju/core/status"
+	"time"
+)
+
+// StatusID represents the status of an entity.
+type StatusID interface {
+	UnsetStatusType | CloudContainerStatusType | UnitAgentStatusType | UnitWorkloadStatusType
+}
+
+// StatusInfo holds details about the status of an entity.
+type StatusInfo[T StatusID] struct {
+	Status  T
+	Message string
+	Data    []byte
```

Comment:
Why slice of bytes and not string? They're both stores as `TEXT` in sql

Reply:
Because the payload isn't a string, it's a series of bytes.

Reply:
Also, everything is a string in sqlite, which to your thought process, would mean everything should be a string.